snap/validate.go

```diff
@@ -468,8 +476,11 @@ func ValidateLayout(layout *Layout, blacklist []string) error {
 	if !isAbsAndClean(mountPoint) {
 		return fmt.Errorf("layout %q uses invalid mount point: must be absolute and clean", layout.Path)
 	}
+
+	// effective path contains a trailing slash for directory-like mount entries.
+	effectivePath := layout.effectivePath()
 	for i := range blacklist {
-		if strings.HasPrefix(mountPoint, blacklist[i]) {
+		if (strings.HasSuffix(blacklist[i], "/") && strings.HasPrefix(effectivePath, blacklist[i])) || effectivePath == blacklist[i] {
```

Comment:
Minor suggestion (maybe it'll be useful later on):
```
// Conflicts checks whether this layout is in conflict with other layout
func (layout *Layout) Conflicts(other *Layout) bool {
...
}

...
for _, other := range blacklist {
    if layout.Conflicts(other) {
        return fmt.Errorf("...")
    }
}
```

Reply:
Yeah, I have something similar since my discussion with. @chipaca yesterday. I'll tweak and push.